### PR TITLE
Feature/add get roles apis

### PR DESCRIPTION
**Pull request by tora-kozic**
Adds GET and GET ALL methods for roles, right now called `get_available_roles` and `get_role`.  

The diff looks a little funny because I merged some changes in from the `main` branch into the `feature/cli` branch, relevant changes are in the `user` client related files

CLI commands for these endpoints will be done in a separate PR

### CODE REVIEW

File: incydr/_users/client.py [hunk 1]
```diff
@@ -170,36 +171,72 @@ def iter_all(
     def get_roles(
         self,
         user_id: str,
-    ) -> List[Role]:
+    ) -> List[UserRole]:
```
Comment (tora-kozic) [inline incydr/_users/client.py:174]: should `get_roles` be changed to `get_user_roles`? (Then `get_available_roles` could be the new `get_roles` too)

Comment (timabrmsn) [reply incydr/_users/client.py:174]: yeah, i like that change

Comment (tora-kozic) [reply incydr/_users/client.py:174]: oh true, that would probably be a little more consistent

Comment (tora-kozic) [reply incydr/_users/client.py:174]: I actually went with `list_user_roles()` for this rename

**Review by timabrmsn — APPROVED** [review]

**Pull request merged**